cmd/cloudFoundryCreateService.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func cloudFoundryCreateService(config cloudFoundryCreateServiceOptions, telemetryData *telemetry.CustomData) {
+
+	cf := cloudfoundry.CFUtils{Exec: &command.Command{}}
+
+	var c = cf.Exec
+
+	//var c command.ExecRunner = &command.Command{}
+
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	var err, logouterr error
+
+	loginOptions := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: config.CfAPIEndpoint,
+		CfOrg:         config.CfOrg,
+		CfSpace:       config.CfSpace,
+		Username:      config.Username,
+		Password:      config.Password,
+	}
+
+	err = cf.Login(loginOptions)
+	if err == nil {
+		err = runCloudFoundryCreateService(&config, telemetryData, c)
+	}
+	if err != nil {
+		logouterr = cf.Logout()
+		if logouterr != nil {
+			log.Entry().WithError(logouterr).Fatal("step execution failed")
+		}
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+	logouterr = cf.Logout()
```

Comment:
Here is a second logout. Would it be possible to re-arrange the control flow so that we have only one cf logout. I believe this ends up in better readable and maintainble code.

Reply:
Could logout be handled via defer?
```
defer func() {
	logoutErr := cfUtils.Logout()
	if logoutErr != nil && returnedError == nil {
		returnedError = fmt.Errorf("Error while logging out occured: %w", logoutErr)
	}
}()
```

Reply:
I implemented it and tested it with defer. Should be better readable and maintainable now. Could you re-check?

Reply:
The defer is mis-placed. Needs to be right after the `err` check for the login.